src/common/components/focus-component.tsx

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { VisualizationConfiguration } from 'common/configs/visualization-configuration';
+import { VisualizationStoreData } from 'common/types/store-data/visualization-store-data';
+import { WindowUtils } from 'common/window-utils';
+import { TabStopRequirementActionMessageCreator } from 'DetailsView/actions/tab-stop-requirement-action-message-creator';
+import * as React from 'react';
+
+export interface FocusComponentProps {
+    windowUtils: WindowUtils;
+    configuration: VisualizationConfiguration;
+    visualizationStoreData: VisualizationStoreData;
+    tabStopRequirementActionMessageCreator: TabStopRequirementActionMessageCreator;
+}
+
+export class FocusComponent extends React.Component<FocusComponentProps> {
+    private static readonly focusEventName = 'focus';
+
+    public componentDidMount(): void {
+        this.props.windowUtils.addEventListener(
+            this.props.windowUtils.getWindow(),
+            FocusComponent.focusEventName,
+            this.handleFocusEvent,
+            false,
+        );
+    }
+
+    public componentWillUnmount(): void {
+        this.props.windowUtils.removeEventListener(
+            this.props.windowUtils.getWindow(),
+            FocusComponent.focusEventName,
+            this.handleFocusEvent,
+            false,
+        );
+    }
+
+    private handleFocusEvent = () => {
+        const tabbing = this.props.configuration.getStoreData(
```

Comment:
In `AdhocTabStopsTestView` we call `getStoreData` on line 99 before instantiating these components; is it possible to reuse the `scanData` variable from there?

https://github.com/microsoft/accessibility-insights-web/blob/18832555826b0666423c956529749ba6b05cd40b/src/DetailsView/components/adhoc-tab-stops-test-view.tsx#L99 

I'm curious about the `checked` value passed to `Toggle` on line 129 and whether this is similar:

https://github.com/microsoft/accessibility-insights-web/blob/18832555826b0666423c956529749ba6b05cd40b/src/DetailsView/components/adhoc-tab-stops-test-view.tsx#L129

Reply:
Would reusing that value work? I'm requerying for the state in the focus handler because I figured that the handler will be called separately/ more often than the code for rendering this component, is that incorrect?

Reply:
I agree the handler will be called separately/more often; I think that is OK. It feels like reusing the value would work because `getStoreData` is a pure function and [relies fully](https://github.com/microsoft/accessibility-insights-web/blob/e3995ceaf8f0161ebcc88c63f56775d09834eac3/src/ad-hoc-visualizations/tab-stops/visualization.tsx#L33) on `props.visualizationStoreData.tests` - and if `props.visualizationStoreData.tests` changes, I'm imagining AdhocTabStopsTestView will re-compute line 99 with its new value.